### PR TITLE
[INJIMOB-967]: selection of credential types

### PR DESCRIPTION
**Pull request by vijay151096**
## Description
selection of credential types, if the issuers supports multiple credential download

## Issue ticket number and link
[INJIMOB-967](https://mosip.atlassian.net/browse/INJIMOB-967)

## Screen Reference

https://github.com/mosip/inji/assets/94220135/16fba495-16b0-46a5-8714-4c7ac7cb47a3

[INJIMOB-941]: https://mosip.atlassian.net/browse/INJIMOB-941?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[INJIMOB-967]: https://mosip.atlassian.net/browse/INJIMOB-967?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: shared/openId4VCI/Utils.ts [hunk 1]
```diff
@@ -111,6 +108,14 @@ export const getBody = async context => {
   };
 };
 
+export const getCredentialType = (context: any) => {
+  return context.selectedCredentialType?.credential_definition?.type
+    ? context.selectedCredentialType.credential_definition.type
+    : context.selectedIssuer?.credential_type
+    ? context.selectedIssuer.credential_type
+    : ['VerifiableCredential', 'MOSIPVerifiableCredential'];
```
Comment (swatigoel) [inline shared/openId4VCI/Utils.ts:116]: Extract this as const defaultCredentialType

File: shared/VCMetadata.ts [hunk 2]
```diff
@@ -82,7 +82,7 @@ export function parseMetadatas(metadataStrings: object[]) {
   return metadataStrings.map(o => new VCMetadata(o));
 }
 
-export const getVCMetadata = context => {
+export const getVCMetadata = (context: any) => {
```
Comment (swatigoel) [inline shared/VCMetadata.ts:85]: Can we have type as `object` instead of `any`?

File: screens/Issuers/CredentialTypeSelectionScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        </Text>
+        <View style={Theme.IssuersScreenStyles.issuersContainer}>
+          <FlatList
+            data={controller.credentialTypes}
+            numColumns={1}
+            keyExtractor={item => item.id}
+            renderItem={({item}) => {
+              return (
+                <CredentialType
+                  testID={removeWhiteSpace(item.id)}
```
Comment (swatigoel) [inline screens/Issuers/CredentialTypeSelectionScreen.tsx:48]: Id is Issuer id eg: esignet, sunbird?

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: screens/Issuers/CredentialTypeSelectionScreen.tsx (hunk 3 above)
Comment (vijay151096) [reply screens/Issuers/CredentialTypeSelectionScreen.tsx:48]: No @swatigoel , Its InsuranceCredential or MosipVerifiableCredential

File: shared/openId4VCI/Utils.ts (hunk 1 above)
Comment (vijay151096) [reply shared/openId4VCI/Utils.ts:116]: The Credential Type **['VerifiableCredential', 'MOSIPVerifiableCredential']**  is used nowhere else except this utility method. extracting it a const will introduce new variable which is unnecessary, so Its better inline. and the method name "getCredentialType" itself suggest its a credential type.

**Pull request merged**